pkg/util/watch/sync.go

```diff
@@ -0,0 +1,108 @@
+package watch
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"golang.org/x/net/context"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	"k8s.io/kubernetes/pkg/api/rest"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/storage"
+	kutil "k8s.io/kubernetes/pkg/util"
+	"k8s.io/kubernetes/pkg/watch"
+
+	"github.com/golang/glog"
+)
+
+// WatchForResourceVersions watches for the given resourceVersion, up to the specified timeout.
+// If no top-level error is encountered, a list of backend errors is returned, with indexes matching the provided watchers.
+// Successful watchers have a nil error at their index. Unsuccessful watchers have a non-nil error at their index.
+func WatchForResourceVersions(versioner storage.Versioner, watchers []rest.Watcher, resourceVersion string, timeout time.Duration) (backendErrors []error, err error) {
```

Comment:
Please avoid named return values.  I'm not seeing where `err` is set, but I can't find it and I'm not certain.